### PR TITLE
Refactor UpdateEIP1559Transaction Component

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Description**
There is very similar code spread across 5 components. The aim of this PR is to reduce the components to use as much common shared code as possible. This PR addresses the UpdateEIP1559Transaction screen. The changes in this PR shouldn't modify existing behaviour when QA'd. Check

Issue

Progresses
https://github.com/MetaMask/mobile-planning/issues/272

**Screenshots/Recordings**

https://user-images.githubusercontent.com/29962968/183502389-cd07caf4-d07d-408a-a8e5-9053dcb53dcd.mov

**Issue**

Progresses https://github.com/metamask/mobile-planning/issues/272

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/core/gasPolling.ts [hunk 1]
```diff
@@ -192,23 +234,36 @@ export const getLegacyTransactionData = ({
...
+        suggestedGasPrice,
+      },
+    },
+    { onlyGas },
+  );
   return parsedTransationData;
 };
 
 /**
  *
```
Comment (sethkfman) [inline app/core/gasPolling.ts:257]: Add a comment for the function.

File: app/core/gasPolling.ts [hunk 2]
```diff
@@ -33,26 +31,36 @@ export const stopGasPolling = () => {
...
+      state.engine.backgroundState.CurrencyRateController.currentCurrency,
+      state.engine.backgroundState.CurrencyRateController.nativeCurrency,
+      state.engine.backgroundState.AccountTrackerController.accounts,
+      state.engine.backgroundState.TokenBalancesController.contractBalances,
+      state.engine.backgroundState.NetworkController.provider.ticker,
+      state.transaction,
+      state.transaction.selectedAsset,
+      state.settings.showCustomNonce,
+    ],
+    shallowEqual,
```
Comment (sethkfman) [inline app/core/gasPolling.ts:62]: What is the reason for using `shallow`?

File: app/components/UI/EditGasFee1559Update/index.tsx [hunk 3]
```diff
@@ -0,0 +1,763 @@
...
+import { useAppThemeFromContext, mockTheme } from '../../../util/theme';
+import { EditGasFee1559UpdateProps } from './type';
+import createStyles from './styles';
+
+const GAS_LIMIT_INCREMENT = new BigNumber(1000);
+const GAS_INCREMENT = new BigNumber(1);
+const GAS_LIMIT_MIN = new BigNumber(21000);
+const GAS_MIN = new BigNumber(0);
+
+const EditGasFee1559Update = ({
```
[the remark this answers is not in the document]
Comment (blackdevelopa) [reply app/components/UI/EditGasFee1559Update/index.tsx:34]: I think we will need to bring this up with the dev team to agree on it. Or we can let it remain `EditGasFee1559Update` until the other dependent screens are fully refactored before the switch

File: app/components/UI/UpdateEIP1559Tx/index.tsx [hunk 4]
```diff
@@ -85,6 +69,37 @@ interface Props {
   onSave: (tx: any) => void;
 }
 
+interface GasTxnProp {
```
[the remark this answers is not in the document]
Comment (blackdevelopa) [reply app/components/UI/UpdateEIP1559Tx/index.tsx]: I updated this in the `GasPolling/types.ts` file. `UseGasTransactionProps` extends `GasTransactionProps (gasData)`

File: app/components/UI/EditGasFee1559Update/types.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  selectedGasValue: string;
+  /**
+   * Initial suggested gas limit. Default is 21000.
+   * @default 21000
+   */
+  initialSuggestedGasLimit: string;
+  /**
+   * Gas fee data for the selected gas value.
+   */
+  gasOptions: {
```
Comment (blackdevelopa) [inline app/components/UI/EditGasFee1559Update/types.ts:14]: This is repeated. Will update this

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by chrisleewilcox** [comment]
Works for test networks...

- Rinkeby
- Groeli

Not able to see transactions for

- Binance
- Polygon

**Comment by blackdevelopa** [comment]
Hey Chris, can we have a sync on this?

**Comment by chrisleewilcox** [comment]
"Submitted" transactions are not updating to "Completed" when blockchain shows success.  When transaction is in this stated the "Speed Up" and "Cancel" will always show popup for retry but retry fails because transactions is already completed.
https://recordit.co/PrmH6nkGiB

**Comment by blackdevelopa** [comment]
I am working with the stability team (Tomas) on this. It seems to happen on production (bnb network) as well.

**Comment by chrisleewilcox** [comment]
I am only seeing the submitted transactions to "Speed Up" or "Cancel" on Rinkeby.  I am not seeing transactions that are submitted but have not been completed and being able to speed up or cancel.

Expected that we can see submitted transactions on all networks before transactions are completed.

**Comment by chrisleewilcox** [comment]
Expectation is that this refactor needs to work on all supported networks.  Sending this back to dev to complete development for this refactor on all supported networks we list in the Add popular networks modal.

**Comment by chrisleewilcox** [comment]
This is not ready for QA.  In order to properly test this transactions on all supported blockchain networks need to be working as expected.  Currently we have issues with transactions on non-ETH/Rinkeby networks.

**Comment by chrisleewilcox** [comment]
Can you provide a recording for the other networks just like you did for Rinkeby?

**Comment by blackdevelopa** [comment]
Polygon Network
https://user-images.githubusercontent.com/29962968/189893719-850c2be6-287d-4e09-b742-0af514204a4d.mov

Mainnet
https://recordit.co/T3hPIobr2k

File: app/components/UI/EditGasFee1559Update/types.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  updateOption:
+    | {
+        isCancel: boolean;
+        maxFeeThreshold: string;
+        maxPriortyFeeThreshold: string;
+        showAdvanced: boolean | undefined;
+      }
+    | undefined;
+}
+export interface EditGasFee1559UpdateProps {
```
Comment (sethkfman) [inline app/components/UI/EditGasFee1559Update/types.ts:13]: Make sure to change EditGasFee1559UpdateProps back to EditGasFee1559Props before closing the PR

**Comment by chrisleewilcox** [comment]
We need to agree on a QA acceptance criteria on this.  I am expecting that this is proven to work on all supported networks as well as at least one custom EIP1559 network and I believe that all this falls in development work.

Referring back to development and either provide evidence of working on all supported networks and one custom network or we discuss further again which is just delaying this.  The development coverage is not sufficient.  Please confirm that the refactor is working on all supported networks here...
image

File: app/components/UI/EditGasFee1559Update/types.ts (hunk 6 above)
Comment (blackdevelopa) [reply app/components/UI/EditGasFee1559Update/types.ts:13]: Should this be done on this PR or when the last refactoring is passed?

**Review by chrisleewilcox — APPROVED** [review]
Approving to see if this run the audit checks again.

**Pull request merged**